.azure-pipelines/merge.yml

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Contributors to the ODPi Egeria project.
+---
+trigger:
+  branches:
+    include:
+      - master
+      - egeria-release-*
+pr: none
+
+stages:
+  - stage: Merge
+    jobs:
+      - job: Deploy
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+          - task: DownloadSecureFile@1
+            name: mavenSettings
+            inputs:
+              secureFile: egeria-artifactory-settings.xml
+          - script: |
+              mkdir -p $(Agent.HomeDirectory)/.m2/
+              mv $(mavenSettings.secureFilePath) $(Agent.HomeDirectory)/.m2/settings.xml
+          - task: ArtifactoryMaven@1
+            inputs:
+              mavenPomFile: 'pom.xml'
+              javaHomeOption: 'JDKVersion'
+              jdkVersionOption: '1.8'
+              jdkArchitectureOption: 'x64'
+              artifactoryResolverService: 'ODPiArtifactory'
+              targetResolveReleaseRepo: 'egeria-staging'
+              targetResolveSnapshotRepo: 'egeria-snapshot'
+              artifactoryDeployService: 'ODPiArtifactory'
+              targetDeployReleaseRepo: 'egeria-staging'
+              targetDeploySnapshotRepo: 'egeria-snapshot'
+              collectBuildInfo: true
+              buildName: 'Egeria'
+              buildNumber: '$(Build.BuildNumber)'
+              includeEnvVars: true
+              goals: 'clean install'
+              options: >-
+                  --batch-mode
+                  -e
+                  -Dmerge
+                  -s $(Agent.HomeDirectory)/.m2/settings.xml
+                  -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+                  -P artifactory
+          - task: ArtifactoryPublishBuildInfo@1
+            inputs:
+              artifactoryService: ODPiArtifactory
+              buildName: 'Egeria'
+          - publish: target
```

Comment:
So far we have only published maven artifacts as part of our build process.
The build process is generating some .gz archives which represent an archive of the runtime code or source - but hasn't yet packaged them up into maven (something I'm adding).

The reason for this As I mentioned above is to use within the docker build for deployment.

However an alternative could be to post other artifacts to jfrog but I'm unsure what standards might apply there? Open to any new suggestions though we should take on as a new issue, the PR is fine for now!

Reply:
Artifactory seems pretty flexible, and you can publish any binary to a 'generic' repo - basically an S3 bucket or site repo in Nexus. I think I saw something about skiping source jars? Are these gzips different then those? Or maybe there's some drawback to generating source jars I don't know.

Reply:
the .gz files represent everything you might need to copy to a new machine to run the code for example
 - the core server runtime
 - the runtime & sources
 - client only libraries.

They are a little distinct from the maven artifacts which are more geared to other java applications using our code (client or more)

As a twist we also have an assembly maven artifact which itself includes the .gz for distribution . (well it would if it worked). I'm not sure how useful it is, but it's been used in lieu of having anywhere else we can post them.

So I think the question here for the project will be to consider where we wish/can publish too, and at that point we'll have to come back to refine the pipelines.

I'm not sure of the source jar reference?

Reply:
> generating some .gz archives which represent an archive of the runtime code or source
I thought these might be the same as `org.odpi.egeria-...-source.jar` files, but they're obviously a bit more than that.

Assuming those files (the .gz) wouldn't be used outside the container, we could generate and store them in the pipeline artifacts as a step before the docker builds. Or even if they were used outside, we could setup a 'generic' Artifactory repo to store them in - whichever makes the most sense.

Reply:
ok thanks. will see how the first stage goes and then open up the docker refinement discussion.